### ISSUE TITLE
still after training not learned some dialogues

### ISSUE THREAD

**Issue by saurabhbidwai** [samtecspg/articulate#302]
0.9.4
windows 10
local 
chorme

![2018-08-12](https://user-images.githubusercontent.com/22415674/44003111-1981f8e6-9e6b-11e8-9257-388a9044b6ea.png)

still after training not learned some dialogues

**Comment by wrathagom** [comment]
@saurabhbidwai I'm a little confused between this issue and #301 exactly the state of your system and agent. There are a lot of errors on the other issue that suggest something wasn't able to be trained. Which if that is the case could cause problems here as well.

For starters looking at your screenshots I can tell that `sausage pizza` and `pineapple pizza` are labeled correctly as the highlighted entity area includes the space, which it shouldn't.

So to get on the same page let's try to fix things in this order:

1. `docker-compose down` This should stop everything that is running and remove the containers
2. `docker-compose up -d` To restart everything
3. Delete and re-create the two examples I mentioned above.
4. Click update to re-train (keep in mind on the version you are on you cannot leave the page until training is completed).
5. Try your example again.

If things aren't working correctly. Please include the logs `docker-compose logs` in your response and the `/agent/{id}/parse` results for cheese.

**Comment by wrathagom** [comment]
It stills looks like your training data has mis-aligned entities.

```
Misaligned entity annotation in
sentence 'cheese with macroni'. Make sure the start and end values of the
annotated training examples end at token boundaries (e.g. don't include
trailing whitespaces).
```

Can you export your agent and provide it to us. We'll troubleshoot and provide feedback.

You can export your agent by going to http://localhost:7500/documentation and then navigating under agent > export.

**Comment by saurabhbidwai** [comment]
{ "language": "en", "timezone": "UTC", "agentName": "Pizza Bot", "useWebhook":
false, "description": "order your pizza", "extraTrainingData": false, "
fallbackResponses": [ "Sorry can you rephrase that?", "I'm still learning
to speak with humans. What you mean?" ], "domainClassifierThreshold": 0.35,
"entities": [ { "regex": null, "uiColor": "#3f51b5", "entityName":
"Toppings", "examples": [ { "value": "Pepperoni", "synonyms": [ "Pepperoni"
] }, { "value": "Sausage", "synonyms": [ "Sausage" ] }, { "value": "Ham", "
synonyms": [ "Ham" ] }, { "value": "Pineapple", "synonyms": [ "Pineapple" ]
}, { "value": "cheese", "synonyms": [ "cheese" ] }, { "value": "macroni", "
synonyms": [ "macroni" ] } ] } ], "domains": [ { "model":
"2740d2a9-cbda-5f62-bcd3-0dd7d9da0bf3", "enabled": true, "domainName": "Order
Receiver", "lastTraining": "Sun Aug 12 2018 17:12:55 GMT+0000 (UTC)", "
intentThreshold": 0.5, "extraTrainingData": false, "intents": [ { "
intentName": "greeting", "useWebhook": false, "examples": [ { "userSays":
"hola", "entities": [] }, { "userSays": "hey", "entities": [] }, { "userSays":
"hello", "entities": [] }, { "entities": [], "userSays": "hi" } ], "scenario":
{ "slots": [], "scenarioName": "greeting", "intentResponses": [ "Hello, My
name is Myzza, How can I help you?" ] } }, { "intentName": "thanks", "
useWebhook": false, "examples": [ { "userSays": "shukran", "entities": []
}, { "userSays": "gracious", "entities": [] }, { "userSays": "great", "
entities": [] }, { "entities": [], "userSays": "ok" }, { "userSays": "okay",
"entities": [] }, { "userSays": "excellent", "entities": [] } ], "scenario":
{ "slots": [], "scenarioName": "thanks", "intentResponses": [ "Thank you,
Happy to help", "Thank you" ] } }, { "intentName": "address", "useWebhook":
false, "examples": [ { "userSays": "9860863729", "entities": [ { "end": 10,
"start": 0, "value": "9860863729", "entity": "sys.duckling_phone-number", "
extractor": "system" } ] }, { "entities": [], "userSays": "road" }, { "
userSays": "society", "entities": [] }, { "entities": [], "userSays":
"apartment" }, { "entities": [], "userSays": "inn" }, { "userSays": "villa",
"entities": [] }, { "entities": [], "userSays": "square" }, { "entities":
[], "userSays": "house" }, { "entities": [], "userSays": "street" } ], "
scenario": { "scenarioName": "address", "slots": [ { "entity":
"sys.duckling_phone-number", "isList": false, "slotName":
"sys.duckling_phone-number", "isRequired": true, "textPrompts": [ "May I
know your contact number?" ] } ], "intentResponses": [ "okay, our delivery
boy would be there soon" ] } }, { "useWebhook": false, "intentName":
"toppings", "examples": [ { "userSays": "i would like to have cheese pizza",
"entities": [ { "end": 27, "value": "cheese", "start": 21, "entity":
"Toppings" } ] }, { "userSays": "pineapple pizza", "entities": [ { "end": 9,
"start": 0, "value": "pineapple", "entity": "Toppings" } ] }, {
"userSays": "sausage
pizza", "entities": [ { "end": 7, "start": 0, "value": "sausage", "entity":
"Toppings" } ] }, { "userSays": "cheese", "entities": [ { "end": 6, "start":
0, "value": "cheese", "entity": "Toppings" } ] }, { "userSays": "sausage", "
entities": [ { "end": 7, "value": "sausage", "start": 0, "entity":
"Toppings" } ] }, { "userSays": "pineapple", "entities": [ { "end": 9, "
start": 0, "value": "pineapple", "entity": "Toppings" } ] }, { "userSays":
"macroni", "entities": [ { "end": 7, "value": "macroni", "start": 0, "entity":
"Toppings" } ] }, { "userSays": "ham", "entities": [ { "end": 3, "value":
"ham", "start": 0, "entity": "Toppings" } ] }, { "userSays": "can i order a
pizza?", "entities": [] }, { "userSays": "i would like to have pizza", "
entities": [] }, { "userSays": "give me pizza", "entities": [] }, { "
userSays": "i want pizza", "entities": [] }, { "userSays": "ham cheese
pizza", "entities": [ { "end": 3, "value": "ham", "start": 0, "entity":
"Toppings" }, { "end": 10, "start": 4, "value": "cheese", "entity":
"Toppings" } ] }, { "userSays": "cheese pizza", "entities": [ { "end": 6, "
value": "cheese", "start": 0, "entity": "Toppings" } ] }, { "userSays": "ham
pizza", "entities": [ { "end": 3, "start": 0, "value": "ham", "entity":
"Toppings" } ] } ], "scenario": { "scenarioName": "toppings", "slots": [ { "
entity": "Toppings", "isList": true, "slotName": "Toppings", "isRequired":
true, "textPrompts": [ "what kind of toppings would you like? we have ham,
cheese, macroni." ] } ], "intentResponses": [ "{{slots.Toppings.value}}
pizza will be on its way! may I know your delivery address and contact
number?" ] } } ] } ] }

**Comment by wrathagom** [comment]
@saurabhbidwai that came through with multiple JSON errors, can you post it as a regular reply rather than an e-mail reply. E-mail replies don't allow formatting on Github.

**Comment by wrathagom** [comment]
```
{
  "language":"en",
  "timezone":"UTC",
  "agentName":"Pizza Bot",
  "useWebhook":false,
  "description":"order your pizza",
  "extraTrainingData":false,
  "fallbackResponses":[
    "Sorry can you rephrase that?",
    "I'm still learning
to speak with humans. What you mean?"
  ],
  "domainClassifierThreshold":0.35,
  "entities":[
    {
      "regex":null,
      "uiColor":"#3f51b5",
      "entityName":"Toppings",
      "examples":[
        {
          "value":"Pepperoni",
          "synonyms":[
            "Pepperoni"
          ]
        },
        {
          "value":"Sausage",
          "synonyms":[
            "Sausage"
          ]
        },
        {
          "value":"Ham",
          "synonyms":[
            "Ham"
          ]
        },
        {
          "value":"Pineapple",
          "synonyms":[
            "Pineapple"
          ]
        },
        {
          "value":"cheese",
          "synonyms":[
            "cheese"
          ]
        },
        {
          "value":"macroni",
          "synonyms":[
            "macroni"
          ]
        }
      ]
    }
  ],
  "domains":[
    {
      "model":"2740d2a9-cbda-5f62-bcd3-0dd7d9da0bf3",
      "enabled":true,
      "domainName":"Order Receiver",
      "lastTraining":"Sun Aug 12 2018 17:12:55 GMT+0000 (UTC)",
      "intentThreshold":0.5,
      "extraTrainingData":false,
      "intents":[
        {
          "intentName":"greeting",
          "useWebhook":false,
          "examples":[
            {
              "userSays":"hola",
              "entities":[

              ]
            },
            {
              "userSays":"hey",
              "entities":[

              ]
            },
            {
              "userSays":"hello",
              "entities":[

              ]
            },
            {
              "entities":[

              ],
              "userSays":"hi"
            }
          ],
          "scenario":{
            "slots":[

            ],
            "scenarioName":"greeting",
            "intentResponses":[
              "Hello, My name is Myzza, How can I help you?"
            ]
          }
        },
        {
          "intentName":"thanks",
          "useWebhook":false,
          "examples":[
            {
              "userSays":"shukran",
              "entities":[

              ]
            },
            {
              "userSays":"gracious",
              "entities":[

              ]
            },
            {
              "userSays":"great",
              "entities":[

              ]
            },
            {
              "entities":[

              ],
              "userSays":"ok"
            },
            {
              "userSays":"okay",
              "entities":[

              ]
            },
            {
              "userSays":"excellent",
              "entities":[

              ]
            }
          ],
          "scenario":{
            "slots":[

            ],
            "scenarioName":"thanks",
            "intentResponses":[
              "Thank you, Happy to help",
              "Thank you"
            ]
          }
        },
        {
          "intentName":"address",
          "useWebhook":false,
          "examples":[
            {
              "userSays":"9860863729",
              "entities":[
                {
                  "end":10,
                  "start":0,
                  "value":"9860863729",
                  "entity":"sys.duckling_phone-number",
                  "extractor":"system"
                }
              ]
            },
            {
              "entities":[

              ],
              "userSays":"road"
            },
            {
              "userSays":"society",
              "entities":[

              ]
            },
            {
              "entities":[

              ],
              "userSays":"apartment"
            },
            {
              "entities":[

              ],
              "userSays":"inn"
            },
            {
              "userSays":"villa",
              "entities":[

              ]
            },
            {
              "entities":[

              ],
              "userSays":"square"
            },
            {
              "entities":[

              ],
              "userSays":"house"
            },
            {
              "entities":[

              ],
              "userSays":"street"
            }
          ],
          "scenario":{
            "scenarioName":"address",
            "slots":[
              {
                "entity":"sys.duckling_phone-number",
                "isList":false,
                "slotName":"sys.duckling_phone-number",
                "isRequired":true,
                "textPrompts":[
                  "May I know your contact number?"
                ]
              }
            ],
            "intentResponses":[
              "okay, our delivery boy would be there soon"
            ]
          }
        },
        {
          "useWebhook":false,
          "intentName":"toppings",
          "examples":[
            {
              "userSays":"i would like to have cheese pizza",
              "entities":[
                {
                  "end":27,
                  "value":"cheese",
                  "start":21,
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"pineapple pizza",
              "entities":[
                {
                  "end":9,
                  "start":0,
                  "value":"pineapple",
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"sausage pizza",
              "entities":[
                {
                  "end":7,
                  "start":0,
                  "value":"sausage",
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"cheese",
              "entities":[
                {
                  "end":6,
                  "start":0,
                  "value":"cheese",
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"sausage",
              "entities":[
                {
                  "end":7,
                  "value":"sausage",
                  "start":0,
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"pineapple",
              "entities":[
                {
                  "end":9,
                  "start":0,
                  "value":"pineapple",
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"macroni",
              "entities":[
                {
                  "end":7,
                  "value":"macroni",
                  "start":0,
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"ham",
              "entities":[
                {
                  "end":3,
                  "value":"ham",
                  "start":0,
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"can i order a pizza?",
              "entities":[

              ]
            },
            {
              "userSays":"i would like to have pizza",
              "entities":[

              ]
            },
            {
              "userSays":"give me pizza",
              "entities":[

              ]
            },
            {
              "userSays":"i want pizza",
              "entities":[

              ]
            },
            {
              "userSays":"ham cheese pizza",
              "entities":[
                {
                  "end":3,
                  "value":"ham",
                  "start":0,
                  "entity":"Toppings"
                },
                {
                  "end":10,
                  "start":4,
                  "value":"cheese",
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"cheese pizza",
              "entities":[
                {
                  "end":6,
                  "value":"cheese",
                  "start":0,
                  "entity":"Toppings"
                }
              ]
            },
            {
              "userSays":"ham pizza",
              "entities":[
                {
                  "end":3,
                  "start":0,
                  "value":"ham",
                  "entity":"Toppings"
                }
              ]
            }
          ],
          "scenario":{
            "scenarioName":"toppings",
            "slots":[
              {
                "entity":"Toppings",
                "isList":true,
                "slotName":"Toppings",
                "isRequired":true,
                "textPrompts":[
                  "what kind of toppings would you like? we have ham,
cheese, macroni."
                ]
              }
            ],
            "intentResponses":[
              "{{slots.Toppings.value}} pizza will be on its way! may I know your delivery address and contact number?"
            ]
          }
        }
      ]
    }
  ]
}
```

**Comment by saurabhbidwai** [comment]
{
  "language": "en",
  "timezone": "UTC",
  "agentName": "Pizza Bot",
  "useWebhook": false,
  "description": "order your pizza",
  "extraTrainingData": false,
  "fallbackResponses": [
    "Sorry can you rephrase that?",
    "I'm still learning to speak with humans. What you mean?"
  ],
  "domainClassifierThreshold": 0.35,
  "entities": [
    {
      "regex": null,
      "uiColor": "#3f51b5",
      "entityName": "Toppings",
      "examples": [
        {
          "value": "Pepperoni",
          "synonyms": [
            "Pepperoni"
          ]
        },
        {
          "value": "Sausage",
          "synonyms": [
            "Sausage"
          ]
        },
        {
          "value": "Ham",
          "synonyms": [
            "Ham"
          ]
        },
        {
          "value": "Pineapple",
          "synonyms": [
            "Pineapple"
          ]
        },
        {
          "value": "cheese",
          "synonyms": [
            "cheese"
          ]
        },
        {
          "value": "macroni",
          "synonyms": [
            "macroni"
          ]
        }
      ]
    }
  ],
  "domains": [
    {
      "model": "2740d2a9-cbda-5f62-bcd3-0dd7d9da0bf3",
      "enabled": true,
      "domainName": "Order Receiver",
      "lastTraining": "Sun Aug 12 2018 17:12:55 GMT+0000 (UTC)",
      "intentThreshold": 0.5,
      "extraTrainingData": false,
      "intents": [
        {
          "intentName": "greeting",
          "useWebhook": false,
          "examples": [
            {
              "userSays": "hola",
              "entities": []
            },
            {
              "userSays": "hey",
              "entities": []
            },
            {
              "userSays": "hello",
              "entities": []
            },
            {
              "entities": [],
              "userSays": "hi"
            }
          ],
          "scenario": {
            "slots": [],
            "scenarioName": "greeting",
            "intentResponses": [
              "Hello, My name is Myzza, How can I help you?"
            ]
          }
        },
        {
          "intentName": "thanks",
          "useWebhook": false,
          "examples": [
            {
              "userSays": "shukran",
              "entities": []
            },
            {
              "userSays": "gracious",
              "entities": []
            },
            {
              "userSays": "great",
              "entities": []
            },
            {
              "entities": [],
              "userSays": "ok"
            },
            {
              "userSays": "okay",
              "entities": []
            },
            {
              "userSays": "excellent",
              "entities": []
            }
          ],
          "scenario": {
            "slots": [],
            "scenarioName": "thanks",
            "intentResponses": [
              "Thank you, Happy to help",
              "Thank you"
            ]
          }
        },
        {
          "intentName": "address",
          "useWebhook": false,
          "examples": [
            {
              "userSays": "9860863729",
              "entities": [
                {
                  "end": 10,
                  "start": 0,
                  "value": "9860863729",
                  "entity": "sys.duckling_phone-number",
                  "extractor": "system"
                }
              ]
            },
            {
              "entities": [],
              "userSays": "road"
            },
            {
              "userSays": "society",
              "entities": []
            },
            {
              "entities": [],
              "userSays": "apartment"
            },
            {
              "entities": [],
              "userSays": "inn"
            },
            {
              "userSays": "villa",
              "entities": []
            },
            {
              "entities": [],
              "userSays": "square"
            },
            {
              "entities": [],
              "userSays": "house"
            },
            {
              "entities": [],
              "userSays": "street"
            }
          ],
          "scenario": {
            "scenarioName": "address",
            "slots": [
              {
                "entity": "sys.duckling_phone-number",
                "isList": false,
                "slotName": "sys.duckling_phone-number",
                "isRequired": true,
                "textPrompts": [
                  "May I know your contact number?"
                ]
              }
            ],
            "intentResponses": [
              "okay, our delivery boy would be there soon"
            ]
          }
        },
        {
          "useWebhook": false,
          "intentName": "toppings",
          "examples": [
            {
              "userSays": "i would like to have cheese pizza",
              "entities": [
                {
                  "end": 27,
                  "value": "cheese",
                  "start": 21,
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "pineapple pizza",
              "entities": [
                {
                  "end": 9,
                  "start": 0,
                  "value": "pineapple",
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "sausage pizza",
              "entities": [
                {
                  "end": 7,
                  "start": 0,
                  "value": "sausage",
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "cheese",
              "entities": [
                {
                  "end": 6,
                  "start": 0,
                  "value": "cheese",
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "sausage",
              "entities": [
                {
                  "end": 7,
                  "value": "sausage",
                  "start": 0,
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "pineapple",
              "entities": [
                {
                  "end": 9,
                  "start": 0,
                  "value": "pineapple",
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "macroni",
              "entities": [
                {
                  "end": 7,
                  "value": "macroni",
                  "start": 0,
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "ham",
              "entities": [
                {
                  "end": 3,
                  "value": "ham",
                  "start": 0,
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "can i order a pizza?",
              "entities": []
            },
            {
              "userSays": "i would like to have pizza",
              "entities": []
            },
            {
              "userSays": "give me pizza",
              "entities": []
            },
            {
              "userSays": "i want pizza",
              "entities": []
            },
            {
              "userSays": "ham cheese pizza",
              "entities": [
                {
                  "end": 3,
                  "value": "ham",
                  "start": 0,
                  "entity": "Toppings"
                },
                {
                  "end": 10,
                  "start": 4,
                  "value": "cheese",
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "cheese pizza",
              "entities": [
                {
                  "end": 6,
                  "value": "cheese",
                  "start": 0,
                  "entity": "Toppings"
                }
              ]
            },
            {
              "userSays": "ham pizza",
              "entities": [
                {
                  "end": 3,
                  "start": 0,
                  "value": "ham",
                  "entity": "Toppings"
                }
              ]
            }
          ],
          "scenario": {
            "scenarioName": "toppings",
            "slots": [
              {
                "entity": "Toppings",
                "isList": true,
                "slotName": "Toppings",
                "isRequired": true,
                "textPrompts": [
                  "what kind of toppings would you like? we have ham, cheese, macroni."
                ]
              }
            ],
            "intentResponses": [
              "{{slots.Toppings.value}} pizza will be on its way! may I know your delivery address and contact number?"
            ]
          }
        }
      ]
    }
  ]
}

**Comment by saurabhbidwai** [comment]
shall I import the json which u just posted? @wrathagom

**Comment by wrathagom** [comment]
This is working fine for me. My only guess is that your agent isn't getting trained for some reason. That may be an out of memory problem. Or you're not staying on the page while it finished.

<img width="292" alt="image" src="https://user-images.githubusercontent.com/1779894/44048455-57454dec-9eff-11e8-959e-e3611740fa90.png">

I would recommend upgrading to our latest release as it has better feedback on the failure/success of the training process.

**Comment by wrathagom** [comment]
Yes it may be worth trying to import the JSON I posted. Keep in mind you'll need to give it a new name or delete your existing agent.

**Comment by saurabhbidwai** [comment]
okay, I'll try it right now

**Comment by saurabhbidwai** [comment]
worked fine!! tnx @wrathagom

**Comment by saurabhbidwai** [comment]
yes
